src/spikeinterface/sortingcomponents/peak_pipeline.py

```diff
@@ -265,6 +344,108 @@ def run_peak_pipeline(recording, peaks, nodes, job_kwargs, job_name='peak_pipeli
     return outs
 
 
+
+def _init_peak_pipeline(recording, nodes):
+    # create a local dict per worker
+    worker_ctx = {}
+    worker_ctx['recording'] = recording
+    worker_ctx['nodes'] = nodes
+    worker_ctx['max_margin'] = max(node.get_trace_margin() for node in nodes)
+
+    return worker_ctx
+
+def _compute_peak_pipeline_chunk(segment_index, start_frame, end_frame, worker_ctx):
+    recording = worker_ctx['recording']
+    max_margin = worker_ctx['max_margin']
+    nodes = worker_ctx['nodes']
+
+    recording_segment = recording._recording_segments[segment_index]
```

Comment:
Should we use the non-private method `select_segments`?

Reply:
Yes! better